File: js-pr/56/javascript/code-quality-readability-maintainability/text_analyzer_34.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
function analyzeTextStatistics(text) {
    const words = text.split(' ').filter(word => word.length > 0);
    const wordCount = words.length;
    const characterCount = text.length;
    const sentenceCount = (text.match(/[.!?]/g) || []).length;
    
    const averageWordLength = wordCount > 0 
        ? words.reduce((sum, word) => sum + word.length, 0) / wordCount 
        : 0;
    
    return {
        words: wordCount,
        characters: characterCount,
        sentences: sentenceCount,
        averageWordLength: parseFloat(averageWordLength.toFixed(2))
    };
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/text_analyzer_34.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
def analyze_text_statistics(text):
    words = text.split()
    word_count = len(words)
    character_count = len(text)
    
    sentence_endings = ['.', '!', '?']
    sentence_count = sum(text.count(ending) for ending in sentence_endings)
    
    average_word_length = (
        sum(len(word) for word in words) / word_count 
        if word_count > 0 else 0
    )
    
    return {
        'words': word_count,
        'characters': character_count,
        'sentences': sentence_count,
        'average_word_length': average_word_length
    }
# {/fact}